module_oop/models.py

```diff
@@ -0,0 +1,100 @@
+import random
+import settings
+import exceptions
+
+
+class Enemy:
+    def __init__(self, level, health) -> None:
+        self.level = level
+        self.health = health
+
+    def decrease_health(self) -> None:
+        self.health -= 1
+        if self.health < 1: raise exceptions.EnemyDown
+
+
+    @staticmethod
+    def select_attack():
+        return random.randint(1, 3) # 1-Knight, 2-Thiev, 3-Wizart
+
+    @staticmethod
+    def select_defence() -> int:
+        return random.randint(1, 3)  # 1-Knight, 2-Thiev, 3-Wizart
+
+
+class Player:
+    def __init__(self, name, health, score, result_fight) -> None:
+        self.health = health
+        self.score = score
+        self.name = name
+        self.result_fight = result_fight
+
+
+    def attack(self, enemy: Enemy) -> None:
+        attack = self.select_attack()
+        defence = enemy.select_defence()
+        self.result_fight = self.fight(attack, defence)
+        print('result_fight: ', self.result_fight)
+        if self.result_fight == 'win':
+            enemy.decrease_health()
+            self.score += settings.SCORE_SUCCESS_ATTACK
+            print('YOUR ATTACK IS SUCCESSFUL!', 'enemy_health', enemy.health,
+                  'my_health', self.health)
+        elif self.result_fight == 'loose':
+            print('YOUR ATTACK IS FAILED!', 'enemy_health', enemy.health,
+                  'my_health', self.health)
+        elif self.result_fight == 'draw':
+            print('IT’S A DRAW!', 'enemy_health', enemy.health,
+                  'my_health', self.health)
+
+    def decrease_health(self) -> None:
+        self.health -= 1
+        if self.health <= 0:
+            with open('result.txt', 'a') as f:
+                f.write(str(self.name) + ' has score: ' + str(self.score) + '\n')
+            print('The score of' + str(self.name) + ' is ' + str(self.score))
+            raise exceptions.GameOver
+
+    def defence(self, enemy: Enemy) -> None:
+        attack = enemy.select_attack()
+        defence = self.select_defence()
+        self.result_fight = self.fight(attack, defence)
+        print('result_fight: ', self.result_fight)
+        if self.result_fight == 'win':
+            enemy.decrease_health()
+            print('YOUR DEFENCE IS SUCCESSFUL!', 'enemy_health', enemy.health,
+                  'my_health', self.health)
+        elif self.result_fight == 'loose':
+            self.decrease_health()
+            print('YOUR DEFENCE IS FAILED!', 'enemy_health', enemy.health,
```

Comment:
![image](https://user-images.githubusercontent.com/22391544/233691340-4818358e-566e-4c37-a2f4-3036d2a002c0.png)
```suggestion
            print('YOUR DEFENCE IS FAILED!')
```